utils/src/logging.rs

```diff
@@ -0,0 +1,68 @@
+use std::path::PathBuf;
+
+use directories::{BaseDirs, ProjectDirs};
+use tracing::level_filters::LevelFilter;
+use tracing_appender::{non_blocking::WorkerGuard, rolling::Rotation};
+use tracing_subscriber::{
+    layer::SubscriberExt, util::SubscriberInitExt, EnvFilter, Layer,
+};
+
+pub trait LogsDirectory {
+    fn log_dir(&self) -> Option<PathBuf>;
+}
+
+impl LogsDirectory for ProjectDirs {
+    fn log_dir(&self) -> Option<PathBuf> {
+        if cfg!(target_os = "linux") {
+            self.state_dir().map(|path| path.join("logs"))
```

Comment:
why is this not the same as datadir as it is defined for linux (and macOS)

Reply:
There is amfeature request regarding this. I am simply using their definition to keep to standardizations